package/MDAnalysis/coordinates/FHIAIMS.py

```diff
@@ -0,0 +1,322 @@
+
+
+class FHIAIMSReader(base.SingleFrameReaderBase):
+    """Reader for the FHIAIMS geometry format.
+
+    .. note::
+       Geometry files are single frame-only
+
+    """
+    format = ['IN', 'FHIAIMS']
+    units = {'time': None, 'length': 'Angstrom', 'velocity': None}
+    _Timestep = Timestep
+
+    def _read_first_frame(self):
+        with util.openany(self.filename, 'rt') as fhiaimsfile:
+            relative, positions, velocities, lattice_vectors = [], [], [], []
+            skip_tags = ["#","initial_moment"]
+            oldline=''
+            for line in fhiaimsfile:
+                line = line.strip()    # perhaps not needed
+                if any([line.startswith(tag) for tag in skip_tags] ):
+                    oldline=line
+                    continue
+                if line.startswith("atom"):
+                    positions.append(
+                        list(np.asarray(line.split()[1:-1], np.float32)))
+                    relative.append('atom_frac' in line)
+                    oldline=line
+                    continue
+                if line.startswith("velocity"):
+                    if not 'atom' in oldline:
+                        raise ValueError(
+                            'Non-conforming line (velocity must follow atom): ({0})in FHI-AIMS input file {0}'.format(line, self.filename))
+                    velocities.append(
+                        list(np.asarray(line.split()[1:], np.float32)))
+                    oldline=line
+                    continue
+                if line.startswith("lattice"):
+                    lattice_vectors.append(list(np.asarray(line.split()[1:])))
+                    oldline=line
+                    continue
+                raise ValueError(
+                        'Non-conforming line: ({0})in FHI-AIMS input file {0}'.format(line, self.filename))
+
+            positions = np.asarray(positions, dtype=np.float32)
+            velocities = np.asarray(velocities, dtype=np.float32)
+            relative = np.asarray(relative)
+            if len(velocities) not in [0,len(positions)]:
+                raise ValueError(
+                    'Found incorrect number of velocity tags in the FHI-AIMS file.')
+            if len(lattice_vectors) not in [0, 3]:
+                raise ValueError(
+                    'Found partial periodicity in FHI-AIMS file. This cannot be handled by MDAnalysis.')
+            if len(lattice_vectors) == 0 and any(relative):
+                raise ValueError(
+                    'Found relative coordinates in FHI-AIMS file withut lattice info.')
+
+            self.n_atoms = n_atoms = len(positions)
+            self.ts = ts = self._Timestep(n_atoms, **self._ts_kwargs)
+
+            if len(lattice_vectors)>0:
+                box = np.asarray(lattice_vectors, dtype=np.float32)
+                positions[np.where(relative)] = np.matmul(
+                    positions[np.where(relative)], box)
```

Comment:
I think this can be written simply as
```suggestion
                positions[relative] = np.matmul(positions[relative], box)
```
I really liked the idea of the `relative` bool array.

Reply:
Is the order of the multiplication correct? I assume some broadcasting is needed. 

Do you have a test case for fractional?

Reply:
I thought so.  Added a test anyway, and passes.